app/ide-desktop/lib/client/src/paths.ts

```diff
@@ -35,3 +36,40 @@ export const PROJECT_MANAGER_PATH = path.join(
     // Placeholder for a bundler-provided define.
     PROJECT_MANAGER_IN_BUNDLE_PATH
 )
+
+/** Relative path of Enso Project PM metadata relative to project's root. */
+export const projectMetadataRelative = path.join('.enso', 'project.json')
+
+/** Get the arguments, excluding the initial program name and any electron dev mode arguments. */
+export const clientArguments = getClientArguments()
+
+/** Check if the given path represents the root of an Enso project. This is decided by the presence of Project Manager's metadata.
+ */
+export function isProjectRoot(candidatePath: string): boolean {
+    const project_json_path = path.join(candidatePath, projectMetadataRelative)
+    try {
+        fss.accessSync(project_json_path, fss.constants.R_OK)
+        return true
+    } catch (e) {
+        return false
+    }
+}
+
+/** Decide what are client arguments, @see {@link clientArguments}. */
+function getClientArguments(): string[] {
+    if (electronIsDev) {
+        // Client arguments are separated from the electron dev mode arguments by a '--' argument.
+        const separator = '--'
+        const separatorIndex = process.argv.indexOf(separator)
+        if (separatorIndex === -1) {
+            // If there is no separator, client gets no arguments.
+            return []
+        } else {
+            // Drop everything before the separator.
+            return process.argv.slice(separatorIndex + 1)
+        }
+    } else {
+        // Drop the leading executable name.
+        return process.argv.slice(1)
+    }
```

Comment:
I don't think this function suits this module, as it contains "File system paths used by the application.".

Reply:
Agreed, moved this one.